### ISSUE TITLE
what effect does  length_cutoff have on  the number of daligner jobs (job_ directories), the runtime of daligner, or the size of the L1 las files?

### ISSUE THREAD

**Issue by dgordon562** [PacificBiosciences/FALCON#323]
Hi, Chris,

what effect does  length_cutoff have on  the number of daligner jobs (job_ directories), the runtime of daligner, or the size of the L1 las files?

Similarly, what effect does the -s parameter in
pa_DBsplit_option = -a -x500 -s350
have on these?  (I'm aware that the smaller the -s, the more daligner jobs...but are they faster/smaller daligner jobs?  Are the las files smaller?)

Thanks!
David

**Comment by dgordon562** [comment]
And I'll bet that the number of blocks is proportional to the size of L1 las files, the runtime of each daligner job, but has no effect on the number of daligner jobs?

**Comment by pb-cdunn** [comment]
`-s` for **DBsplit** determines the size of blocks (in MB). Halving `-s` should roughly quadruple the number of **daligner** block-comparisons, and each of those should go 4x as fast. However, a **daligner** *job* is a group of block-comparisons. E.g.

    daligner 1 1 2 3

means "compare 1-to-1, 1-to-2, and 1-to-3". That's 3 different block-comparisons. (Technically, it's 5 with flips 2-to-1 and 3-to-1, and 10 with reverse-complement.) The block-comparisons are grouped by **HPCdaligner** based on `-dal B`, where B is the number of blocks we would like for each job-group. Higher B would make a single job do more comparisons, so it would take longer and produce more .las output. However, the total size of all merged .las data is independent of those settings.

Higher `length_cutoff` means that fewer seed reads will be used. It ends up in the `-H` argument to **daligner**. All else unchanged, that should reduce the sizes of `.las` files (both individually and total) because in the a/b pairs, **daligner** will only dump pairs with an a read above the length_cutoff. CPU time should also drop since instead of O(N*N) comparisons within a block, it will do O(M*N), where M is the number of reads above the threshold, and N is the total number of reads in the DB.

Another way to get smaller/less .las files is to increase `-x` for **DBsplit**. You'd be ignoring shorter reads. It's really a "raw-reads" cutoff.

A better way to get smaller (and smaller total, but not fewer) .las files is to tweak the memory-constraints of **daligner**, using `-t` or `-M`. That would also impact the total memory consumption, and both the individual and total runtimes. By lowering t (ignore kmers with greater count than t) or lowering M (max memory), you'd be telling **daligner** to ignore a greater number of high-frequency kmers.

The filtration settings are mysterious to me. In theory, higher `-k`, lower `-w`, and  lower `-h` would each yield fewer alignments (if `-t` or `-M` is not already causing them to be missed), but it's highly non-linear.

> I'll bet that the number of blocks is proportional to the size of L1 las files, the runtime of each daligner job, but has no effect on the number of daligner jobs?

In summary, it's complicated. And note that the size of each .las and the number of .las files are not very important. The *total* size of .las files is your disk use, and that is (almost) unrelated to block-size. But yes, the runtime of a given **daligner** job is generally proportional to the sum of all .las files in its own output.

**Comment by dgordon562** [comment]
Thanks, Chris!  This is VERY useful information.  I've added it to my private copy of falcon documentation and you might consider adding it to the public falcon documentation as well.  

This is my summary (but your full explanation is more valuable):

DBsplit's -s (block size) has little influence on either total cumulative runtime or total disk space

total disk space and total cumulative runtime *is* affected by length_cutoff and daligner's -M and DBsplit's -x
